crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -102,17 +74,14 @@ where
     T: StateData,
 {
     pub fn on_update(pred: T) -> RunCriteriaDescriptor {
-        let pred_clone = pred.clone();
         (move |state: Res<State<T>>| {
             state.stack.last().unwrap() == &pred && state.transition.is_none()
         })
         .chain(should_run_adapter::<T>)
         .after(DriverLabel::of::<T>())
-        .label_discard_if_duplicate(StateCallback::Update.into_label(pred_clone))
```

Comment:
I'm not immediately sure why this is being removed. And can we remove this method entirely? It looks like you've deleted a bunch of uses of it.

Reply:
This call just adds a label of type `StateRunCriteriaLabel`, which I removed because it requires runtime labels, is private, and isn't used anywhere. All of the checks passed after removing it, so I'm pretty sure this is just dead code that rustc missed.

Reply:
aren't these labels for if someone wants to order around these systems?

Reply:
> aren't these labels for if someone wants to order around these systems?

That was probably the intention, but as far as I can tell the labels were never exposed outside of the module, so they were never actually used for that purpose. The dead code lint got suppressed because the type had traits implemented for it, but in practice it was never used.